airbyte-integrations/connectors/source-slack/source_slack/spec.json

```diff
@@ -4,14 +4,33 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "title": "Slack Spec",
     "type": "object",
-    "required": ["api_token"],
+    "required": ["api_token", "start_date", "lookback_window", "join_channels"],
     "additionalProperties": false,
     "properties": {
       "api_token": {
         "type": "string",
         "title": "API Token",
         "description": "A slack bot token. See the <a href=\"https://docs.airbyte.io/integrations/sources/slack\">docs</a> for instructions on how to generate it.",
         "airbyte_secret": true
+      },
+      "start_date": {
+        "type": "string",
+        "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
+        "description": "UTC date and time in the format 2017-01-25T00:00:00Z. Any data before this date will not be replicated.",
+        "examples": ["2017-01-25T00:00:00Z"],
+        "title": "Start date"
+      },
+      "lookback_window": {
+        "type": "integer",
+        "title": "Lookback window",
+        "description": "Lookback window in days from which to load data.",
+        "examples": [7, 14]
+      },
+      "join_channels": {
```

Comment:
@sherifnada are we sure we want this option configurable by User? I mean what is the Users scenario when don't need to use this functionality?

Reply:
it was requirement of the ticket, to make these options configurable

Reply:
@po3na4skld discussed with @sherifnada, as the case for which we need to disable auto-joining wasn't clear for me. The solution might be suboptimal, but it is sufficient for this ticket. resolving...